### PR TITLE
Move all clusters to be RAPID clusters, and verify them using valid_v…

### PR DESCRIPTION
**Pull request by Obliviour**
…ersions correctly

## Fixes / Features
- Previous was using the non-release version clusters verification list when creating rapid gke clusters.
- Moved GPU clusters to also be rapid based for consistency.

## Testing / Documentation
Create a cluster with the default GKE version: success
Create an additional node pool with default GKE version: success
Specify a GKE version that is valid: success
Specify a GKE version that is invalid: failed correctly

- [ y ] Tests pass
- [ y ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

**Review by RoshaniN — COMMENTED** [review]
Thanks for the change, Victor!

File: xpk.py [hunk 1]
```diff
@@ -4186,15 +4169,15 @@ def get_gke_node_pool_version(
...
   # In rare cases, user's provided gke version may be invalid, but gke will return an error if so.
   # An example scenario is if the user provided gke version is greater than the master version.
   if not is_supported_node_pool_version:
     xpk_print(
         f'Planned node pool version {node_pool_gke_version} is not supported in'
-        ' valid node_pool_gke_versions'
-        f' {gke_server_config.valid_node_versions}Please adjust the gke version'
+        ' valid version'
+        f' {gke_server_config.valid_versions}\nPlease adjust the gke version'
         ' using --gke-version=x or remove the arg and depend on xpk default of'
```
[the remark this answers is not in the document]
Comment (Obliviour) [reply xpk.py:4181]: Does setting --gke-version override the planned version as needed?

Yes this does.

I believe I was not successful in doing so with the previous logic.

Can you try again?

[the remark this answers is not in the document]
Comment (RoshaniN) [reply xpk.py:4181]: I tried again. Same results

Commands - 
```
python3 xpk.py cluster create --cluster $CLUSTER --num-slices=2 --tpu-type=v4-8 --default-pool-cpu-machine-type=n2-standard-32 --on-demand --project $PROJECT --zone $ZONE --enable-pathways --gke-version=1.29.4-gke.1447001
```

Saw the same behavior without `--enable-pathways` also.


```
Recommended GKE Version: 1.30.0-gke.1167000
[XPK] Error: Planned GKE Version 1.29.4-gke.1447001 is not valid.Checks failed: Is Master Valid: False
Is Valid Node Version: True
[XPK] Please select a gke version from the above list using --gke-version=x argument or rely on the default gke version: 1.30.0-gke.1167000
[XPK] XPK failed, error code 1
```

Ideally, it should not check against the recommended gke version if that has been provided through command line.

File: xpk.py [hunk 2]
```diff
@@ -4118,24 +4108,17 @@ def get_gke_control_plane_version(
...
-  )
+  is_valid_version = master_gke_version in gke_server_config.valid_versions
 
-  if not is_valid_master_version or not is_valid_node_version:
+  if not is_valid_version:
     xpk_print(
-        f'Planned GKE Version: {master_gke_version}\n Valid Master'
-        f' Versions:\n{gke_server_config.valid_master_versions}\nValid Node'
-        f' Versions:\n{gke_server_config.valid_node_versions}\nRecommended GKE'
+        f'Planned GKE Version: {master_gke_version}\n Valid Versions'
```
[the remark this answers is not in the document]
Comment (Obliviour) [reply xpk.py:4115]: Planned = the version to be used by the cluster
Valid Versions are any possible version that could be used
Recommnded is the default version that GKE says to use

File: xpk.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RoshaniN) [reply xpk.py:4181]: Synced offline, Victor was able to test with a valid version from the new valid versions list.

**Review by RoshaniN — APPROVED** [review]

**Pull request merged**